lib/qm-dsp/dsp/tempotracking/DownBeat.cpp

```diff
@@ -207,38 +206,7 @@ DownBeat::findDownBeats(const float *audio,
             oldspec[j] = newspec[j];
         }
     }
-
-    // We now have all spectral difference measures in specdiff
-
-    int timesig = m_bpb;
-    if (timesig == 0) timesig = 4;
-
-    d_vec_t dbcand(timesig); // downbeat candidates
-
-    for (int beat = 0; beat < timesig; ++beat) {
-        dbcand[beat] = 0;
-    }
-
-   // look for beat transition which leads to greatest spectral change
-   for (int beat = 0; beat < timesig; ++beat) {
-       int count = 0;
-       for (int example = beat-1; example < (int)m_beatsd.size(); example += timesig) {
-           if (example < 0) continue;
-           dbcand[beat] += (m_beatsd[example]) / timesig;
-           ++count;
-       }
-       if (count > 0) {
-           dbcand[beat] /= count;
-       }
-   }
-
-    // first downbeat is beat at index of maximum value of dbcand
-    int dbind = MathUtilities::getMax(dbcand);
-
-    // remaining downbeats are at timesig intervals from the first
-    for (int i = dbind; i < (int)beats.size(); i += timesig) {
-        downbeats.push_back(i);
-    }
+    return m_beatsd;
```

Comment:
Why return a copy **and** store it as a member? If we store it as a member we can return a const-ref.